### DISCUSSION TITLE
Conditional Product Options / Dependent Options

### DISCUSSION THREAD

**Discussion by mraf** [givanz/Vvveb#116] [Ideas]
Hi,

as some of ecommerce stores have more custom products will be good to have a way to assign one option to another. 

As Vvveb already have options implemented I think good feature will be make those options more advanced by creating customizable options dependence. 

On the product page, the buyer will see various options that are relevant. As choices are made, the next available options will be related to the ones the buyer has previously selected. Also based on previous choise option can be visible or not.

1. It makes the whole process smoother for customers. They only see options relevant to what they've already chosen, which makes selecting items much easier.

2. Customers won't get overwhelmed by too many options at once. They'll go through a step-by-step process, picking exactly what they want without feeling bogged down.

3. With Dependent Options, customers can really make the product their own. They can customize based on color, size, or features, making it feel like it's made just for them.

4. By limiting choices based on what they've already chosen, you help customers avoid picking things that don't go together. This means fewer returns or exchanges later on.

5. You can use this feature to suggest related or premium options based on what they've already picked. It's a great way to encourage customers to explore more and maybe spend a little extra.

This will make options more dynamic too. 

example of implementation based on conditions plus visibility

![image](https://github.com/givanz/Vvveb/assets/10253590/0500a297-2ece-42e8-b3b3-a2d72515b85d)

**Comment by givanz** [comment]
Hi

Thanks for the suggestion and screenshot, this is a good must have feature for advanced product configuration.

I think dependent options and advanced options configuration should be added through a plugin to keep product options from core simpler.

The interface to configure the options can get complex and confusing for most people that need simple options.

This feature will be implemented after ecommerce features like advanced product options filtering and recurring payments/subscriptions.

**Comment by mraf** [comment]
Thanks for making this idea possible! Agree that for most people which using simple products those options are not that important, but for shops which need more advanced selections it will be great. 

Those dependent options could also have another important feature which is **link to product from option** 
This will allow options be linked to simple product by product id or sku. Currently I saw this only in .net ecommerce solutions, but to be honest it it the coolest thing for ecommerce stores.

Heaving this simple products linked with options alows to keep stock of simple products, take price for options or image. 

The simplest example is a configurable desktop computer where customer on shop product page selecting Processor, RAM and HDD. Those parts are linked with options so in warehouse they can be stored as separate parts.

**Comment by givanz** [comment]
This is a useful feature, also the option can be disabled if the linked product is out of stock.
